File: app/src/test/java/com/chetdeva/madrasi/root/landing/LandingInteractorTest.java
package com.chetdeva.madrasi.root.landing;

import com.ubercab.test.UberTestBase;
import com.uber.rib.core.InteractorHelper;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

public class LandingInteractorTest extends UberTestBase {

  @Mock private LandingInteractor.LandingPresenter presenter;
  @Mock private LandingRouter router;

  private LandingInteractor interactor;

  @Before
  public void setup() {
    MockitoAnnotations.initMocks(this);

    interactor = TestLandingInteractor.create(presenter);
  }

  /**
   * TODO: Delete this example and add real tests.
   */
  @Test
  public void anExampleTest_withSomeConditions_shouldPass() {
    // Use InteractorHelper to drive your interactor's lifecycle.
    InteractorHelper.attach(interactor, presenter, router, null);
    InteractorHelper.detach(interactor);

    throw new RuntimeException("Remove this test and add real tests.");
  }

}

File: app/src/androidTest/java/com/chetdeva/madrasi/root/landing/LandingInteractorTest.kt
package com.chetdeva.madrasi.root.landing

import com.uber.rib.core.RibTestBasePlaceholder
import com.uber.rib.core.InteractorHelper

import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.MockitoAnnotations

class LandingInteractorTest : RibTestBasePlaceholder() {

  @Mock internal lateinit var presenter: LandingInteractor.LandingPresenter
  @Mock internal lateinit var router: LandingRouter

  private var interactor: LandingInteractor? = null

  @Before
  fun setup() {
    MockitoAnnotations.initMocks(this)

    interactor = TestLandingInteractor.create(presenter)
  }

  /**
   * TODO: Delete this example and add real tests.
   */
  @Test
  fun anExampleTest_withSomeConditions_shouldPass() {
    // Use InteractorHelper to drive your interactor's lifecycle.
    InteractorHelper.attach<LandingInteractor.LandingPresenter, LandingRouter>(interactor!!, presenter, router, null)
    InteractorHelper.detach(interactor!!)

    throw RuntimeException("Remove this test and add real tests.")
  }
}